### PR TITLE
Cleanup of the CRSF header and code

### PR DESCRIPTION
**Pull request by pkendall64**
This is a maintenance PR, which just shuffles things around to simplify the ifdefs to make the code a little easier to follow.

### CODE REVIEW

**Review by JyeSmith — APPROVED** [review]

**Review by CapnBry — CHANGES_REQUESTED** [review]

File: src/lib/CRSF/CRSF.cpp [hunk 1]
```diff
@@ -15,7 +19,7 @@ portMUX_TYPE FIFOmux = portMUX_INITIALIZER_UNLOCKED;
 RTC_DATA_ATTR int rtcModelId = 0;
 #elif defined(PLATFORM_ESP8266)
 HardwareSerial CRSF::Port(0);
-#elif CRSF_TX_MODULE_STM32
```
[the remark this answers is not in the document]
Comment (pkendall64) [reply src/lib/CRSF/CRSF.cpp:18]: Good call

File: src/lib/CRSF/CRSF.cpp [hunk 2]
```diff
@@ -394,12 +381,12 @@ void ICACHE_RAM_ATTR CRSF::RcPacketToChannelsData() // data is packed as 11 bits
         bitsMerged -= srcBits;
     }
 
+    #if defined(PLATFORM_ESP32)
     if (prev_AUX1 != ChannelData[4])
     {
-    #if defined(PLATFORM_ESP32)
```
[the remark this answers is not in the document]
Comment (pkendall64) [reply src/lib/CRSF/CRSF.cpp:399]: Why don't we always call `devicesTriggerEvent` when the arm state changes? It's only setting a flag for the main loop!
I'd have to make that function `ICACHE_RAM_ATTR` too.

[the remark this answers is not in the document]
Comment (pkendall64) [reply src/lib/CRSF/CRSF.cpp:399]: Move the ifdef back.

[the remark this answers is not in the document]
Comment (CapnBry) [reply src/lib/CRSF/CRSF.cpp:399]: Dangit! I was hoping you'd do some magic instead 🎇

**Review by CapnBry — APPROVED** [review]

**Pull request merged**